src/silx/gui/widgets/FilenameCompleter.py

```diff
@@ -0,0 +1,23 @@
+from pathlib import Path
+
+from .. import qt
+
+
+class FilenameCompleter(qt.QCompleter):
+    """
+    A QCompleter that provides autocompletion for file paths.
+    """
+
+    def __init__(
+        self, parent: qt.QWidget | None = None, root: str | None = None
+    ) -> None:
+        super().__init__(parent=parent)
+
+        completer = qt.QCompleter()
+        model = qt.QFileSystemModel(completer)
+        model.setOption(qt.QFileSystemModel.Option.DontWatchForChanges, True)
+        if root is None:
+            model.setRootPath(Path(__file__).root)
```

Comment:
Maybe good to check how it behaves on Windows

Reply:
Yes. [anchor](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.anchor) might be a better default

Reply:
Tested on Windows and `/` is actually a sensible working default